### PR TITLE
MangaController overhaul

### LINKED ISSUE #5793: UI updating constantly when updating library
### Steps to reproduce

Update library.
Enter into any Manga.
Check download buttons.
Try to long press a chapter.

### Expected behavior

No flickering.

### Actual behavior

Flickering.

### Crash logs

No crash

### Tachiyomi version

0.12.1

### Android version

Android 9

### Device

Mi Mix 2

### Other details

In the video I'm always trying to long-press on a chapter (except when I succeed and it's selected).
Sometimes it works, sometimes you can see as if the long-press is getting interrupted (that's why I assume the UI is refreshing constantly, no idea though)

https://user-images.githubusercontent.com/7734808/130874791-9defe25c-0e8d-42b6-a1af-703a9e0c2cd6.mp4

I know this is probably the same as #5553 and #5064 but I still wanted to open this issue so the dev can merged these issues and have as many info as possible.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.12.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #5888: Manga page top bar not updating when scrolliing
### Steps to reproduce

Scroll in manga info (can't reproduce it reliably, only sometimes)

### Expected behavior

Sometimes transparent when scrolling to top, Sometimes opaque when scrolling

### Actual behavior

(edit:) still reproducible, r3999

Screenshot_20211205-093146
Screenshot_20211205-093158

### Crash logs

_No response_

### Tachiyomi version

r3783

### Android version

Android 10

### Device

Redmi Note 9 Pro

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.12.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #5936: Fastscroll snapping to very top of screen instead of top bar
### Steps to reproduce

Use fastscroll in manga page

### Expected behavior

0.12.1 behavior

https://user-images.githubusercontent.com/63973749/133911743-9a0cfaad-f905-407d-90b8-b9d7339f6247.mp4

### Actual behavior

(edit:) Still reproducible, r3999

Opaque top bar bug (#5888) makes it more obvious

Screenshot_20211205-093146

r3826

https://user-images.githubusercontent.com/63973749/133911760-c97fc106-82c1-419e-83b0-37d03805f3a9.mp4

### Crash logs

_No response_

### Tachiyomi version

r3826

### Android version

A10

### Device

Redmi Note 9 Pro

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.12.2](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #6080: Bad custom cover cache when thumbnail_url isn't unique
### Steps to reproduce

1. Add multiple manga with the same thumbnail url, eg Dynasty on series without a cover.
2. Set custom covers for the manga

### Expected behavior

Each manga displays it's own custom cover

### Actual behavior

It appears correct on first launch, but begins to show incorrect covers after navigation.
Screenshot_1633924669
Screenshot_1633924677
Screenshot_1633924679

### Crash logs

[debug.txt](https://github.com/tachiyomiorg/tachiyomi/files/7319492/debug.txt)

### Tachiyomi version

0.12.3 and git aae011ed8

### Android version

Android 8 and 9

### Device

Kindle Fire HD8 Gen 6 + LineageOS 15.1 and Emulator

### Other details

The detail cover view is also affected, and saving the cover to file saves the bad cache.
Screenshot_1634011555

Trowing a `memoryCachePolicy(CachePolicy.DISABLED)` into MangaFullCoverDialog.setImage fixes this sub-issue:
Screenshot_1634011638

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.12.3](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #6311: Manga page fastscroll hidden behind top bar
### Steps to reproduce

First bug:
Go to manga page
Select a chapter
Deselect a chapter

Second bug:
Go to manga page
Read a chapter
Go back to manga page

### Expected behavior

Fastscroll under top bar
Screenshot_20211205-094606

### Actual behavior

Fastscroll behind top bar
Screenshot_20211205-094618

### Crash logs

_No response_

### Tachiyomi version

(edit:) r4353

### Android version

(edit:) Android 12

### Device

Redmi Note 9 Pro (ArrowOS)

### Other details

Different bug from #6132 

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.12.3](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #6567: Chapter closing animation can not be skipped
### Steps to reproduce

Open a chapter
Close a chapter
Try to go back to the overview while closing animation is shown

### Expected behavior

When leaving a chapter the animation should start, but if you click the back button again while in the animation, the animation should stop and you return to the previous page (which is most likely an overview).

Also the animations should be an option in the settings since it might be slower on older phones

### Actual behavior

While the opening an closing animation when reading a chapter looks nice, it can not be skipped.
Especially chapter closing animation takes nearly one second to end and can not be skipped to leave the manga.

### Crash logs

_No response_

### Tachiyomi version

0.13.1

### Android version

Android 9

### Device

Hisense A6L

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [x] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.13.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #6646: Resume button/option snaps downwards instead of smooth animation
### Steps to reproduce

1• Open any manga/webtoon/something
2• Read a chap by tapping on resume
3• When you press back or swipe to go back, you'll see the resume thingy snapping lower

### Expected behavior

It shouldn't snap. Should be smooth. 

### Actual behavior

It snaps downwards. 

https://user-images.githubusercontent.com/62374996/153816330-1f1b4287-8d2b-4d96-92c1-5b3fef12dc27.mp4

This is in .5x animation speed because that's what I set things to in dev options but I tried and with 1x speed it snaps too. It's just slower. 

### Crash logs

_No response_

### Tachiyomi version

0.13.1

### Android version

Android 11

### Device

OnePlus Nord running OxygenOS 11

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.13.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #6809: Hidden "draggable" bar on chapter lists
### Steps to reproduce

this is the "draggable" bar i am writing about:
IMG_20220323_132722

https://user-images.githubusercontent.com/75091899/159698515-e49bde4a-455d-4209-af39-6b42c92a8758.mp4

1. go to the chapter list of a manga with 800+ chapters (martial peak, apotheosis etc..)
2. scrolling down until the draggable bar shows up

### Expected behavior

The app should show the draggable bar after a few chapters scrolled down.

### Actual behavior

The app shows up the draggable bar ( a size that allows me to drag it comfortably) when the 10-15% of the max amount of chapters is scrolled down.

### Crash logs

_No response_

### Tachiyomi version

0.13.1

### Android version

android 8

### Device

huawei p8 lite 2017

### Other details

This issue is not a problem when there aren't many chapters, but when the manga starts to have a lot of them (700+ chapters, see martial peak with 2000+) it becomes annoying and the user has to scroll down the 10-15% of the max amount of chapters ( 90+ and it is slow) the get the draggable bar to show up and use it to navigate through the vast amount of chapters.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.13.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #7147: Reading Resume FAB causes Last Page Read to jumps pages on exit
### Steps to reproduce

See the video for a random manga I reproduced the issue with. But the issue also occasionally happen on other mangas.
Just by using webtoon mode and scrolling (doesn't matter up or down) just a bit might make the last read page become bugged and jump and skip pages.

Pay close attention to the "Page: #" and notice how it jumps.

https://user-images.githubusercontent.com/39028181/168499685-ac3eaa05-6b53-4523-b33f-25ae655b3371.mp4

**The jump in the number happens exactly the second I press back to close the reader. At the last few frames someone can clearly see that the number of current page index down have suddenly changed on close.**

### Expected behavior

Last page read should calculate correctly

### Actual behavior

Last page read is incorrect and causes the user to miss pages if he didn't notice the jump.

### Crash logs

_No response_

### Tachiyomi version

Last debug build
App version: 0.13.4-4374 (dev, 5dc688dc2e, 81, 2022-05-16T00:15Z)

### Android version

Android 9

### Device

Emulator

### Other details

I was not able to reproduce this bug all the time, I still don't know what triggers it precisely.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.13.4](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #7222: Since r4430 chapter read status not updated
### Steps to reproduce

Have r4430 installed.
Have any series with unread chapters.
Read at least one chapter.
Exit reader (goes back to series description with chapter list)
See the read chapter still showing unread.
You have to change screen and come back for the right read status to show.

Same for at least the "news" tab, the read status is only retrieved when loading the tab (and not real time or when coming back to it from reading a chapter)

### Expected behavior

Read status showing correctly

### Actual behavior

See reproduce steps

### Crash logs

_No response_

### Tachiyomi version

r4430

### Android version

Android 12

### Device

Mi 11t

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.13.4](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by ivaniskandar**
What's changed:
* Compose UI except on dialogs and sheets (fixes #5793 fixes #5888 fixes #5936 fixes #6311 fixes #6809)
* Major cleanups on controller, presenter and all things related

What's missing:
* Start/resume FAB transition since Compose doesn't support it (closes #6567 closes #6646 fixes #7147)

o7

### CODE REVIEW

**Comment by Saud-97** [comment]
> What's missing:
> 
> Start/resume FAB transition since Compose doesn't support it (closes https://github.com/tachiyomiorg/tachiyomi/issues/6567 closes https://github.com/tachiyomiorg/tachiyomi/issues/6646)

Also fixes https://github.com/tachiyomiorg/tachiyomi/issues/7147 since the FAB transition was the cause of the bug and it got removed for now.

I have a comment on the download progress circle animation. It honestly looks bad and really needs to be smoothened out with animation and fading.

I recommend adding
 * Animation on progress start from 0 to the next value
 * Do not change color of the arrow abruptly when progress reaches 50%+, maybe keep it same or make the color fade
 * Animation when progress reaches 100%.

Watch the video to understand what I meant.

https://user-images.githubusercontent.com/39028181/171958192-96ada19d-a0da-4839-8604-b6880fb97d37.mp4

**Comment by arkon** [comment]
Please refrain from commenting on larger nitpicks, we can address those as followups. This PR is too big as-is anyway, but I'll try to review it tomorrow.

File: app/src/main/java/eu/kanade/data/chapter/ChapterRepositoryImpl.kt [hunk 1]
```diff
@@ -11,6 +12,33 @@ class ChapterRepositoryImpl(
     private val databaseHandler: DatabaseHandler,
 ) : ChapterRepository {
 
+    override suspend fun addAll(chapters: List<Chapter>): List<Chapter> {
+        return try {
```
[the remark this answers is not in the document]
Comment (ivaniskandar) [reply app/src/main/java/eu/kanade/data/chapter/ChapterRepositoryImpl.kt:16]: This one's by @AntsyLich but it's similar with the existing approach on handling errs on db updates

**Comment by Soitora** [comment]
### Let me pray for you 🙏🏻 

image

**Comment by ivaniskandar** [comment]
mfw

**Pull request merged**